src/jit/regalloc.cpp

```diff
@@ -5552,6 +5560,10 @@ regMaskTP Compiler::rpPredictAssignRegVars(regMaskTP regAvail)
                         continue;
 
                     noway_assert(argDsc->lvIsParam);
+                    bool isFloat = argDsc->IsFloatRegType();
+#ifdef ARM_SOFTFP
+                    isFloat = false;
+#endif
                     noway_assert(inArgBit & (isFloat ? RBM_FLTARG_REGS : RBM_ARG_REGS));
```

Comment:
I don't think this change is necessary, because raAvoidArgRegMask should never contain any floating-point registers.